src/utils/consolidateStreamedStyles.js

```diff
@@ -1,26 +1,18 @@
 // @flow
-/**
- * When using streaming rendering, style blocks are emitted in chunks directly
- * next to the HTML they reference. In order to prevent errors during rehydration
- * (since React doesn't know about the style blocks we are interleaving) this
- * method relocates all styled-component blocks to the end of `<head>`.
- *
- * NOTE: this method MUST be called before ReactDOM.hydrate().
- */
-export default function consolidateStreamedStyles() {
-  const blocks = Array.from(
-    document.querySelectorAll('style[data-styled-components]')
-  )
-
-  if (blocks.length) {
-    const frag = document.createDocumentFragment()
 
-    for (let i = 0, len = blocks.length; i < len; i += 1) {
-      // $FlowFixMe
-      frag.appendChild(blocks[i].parentNode.removeChild(blocks[i]))
-    }
-
-    // $FlowFixMe
-    document.head.appendChild(frag)
+/* This function is DEPRECATED and will be removed on the next major version release.
+ * It was needed to rehydrate all style blocks prepended to chunks before React
+ * tries to rehydrate its HTML stream. Since the master StyleSheet will now detect
+ * the use of streamed style tags and will perform the rehydration earlier when needed
+ * this function will not be needed anymore */
+export default function consolidateStreamedStyles() {
+  if (process.env.NODE_ENV !== 'production') {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'The consolidateStreamedStyles function for streamed SSR rehydration was removed.\n' +
+        'The rehydration will now be automatically performed at an earlier point in time ' +
+        'on the client so that this function is now not needed anymore and is thus obsolete.\n' +
+        '- Consider removing this call on the client and use React + styled-components as you would expect'
```

Comment:
Slight rephrase of this entire error message:

```
styled-components automatically does streaming SSR rehydration now, so manually calling consolidateStreamedStyles is no longer necessary and a noop now.
- Please remove the consolidateStreamedStyles call from your client.
```

Reply:
Sure thing

Reply:
done